### PR TITLE
fix some buttons are not responding after theme change

### LINKED ISSUE #14780: Some buttons stop responding when changing mode (dark/light) on certain views
# Bug Report

### Reproduction

Case 1:
1. Create a community
2. Open a channel
3. On the channel view change the mode from light to dark
4. Open the community again
5. Try to open the channel

Case 2:
1. Go to profile > Contacts > tap `+ Add new contact`
2. Switch the mode being on this screen
3. Open `Contacts` > tap `+ Add new contact` again

#### Expected behavior
Can change the mode on any screen

#### Actual behavior

Some buttons stop being responsive until relogin

https://user-images.githubusercontent.com/67952253/212380286-b4f8be37-0c5a-46e3-8298-7539d3e3bc25.mp4

### Additional Information

- Status version: nightly 13/01
- Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes: https://github.com/status-im/status-mobile/issues/14780

### Issue Cause
- Chat was not closing on system theme change
- As `:close-chat` and `:navigate-to-chat` both had `:dispatch` key, it was causing `:merging-fx-with-common-keys` error, due to unclosed chat
- Models were not dismissed on theme change

status: ready

### CODE REVIEW

**Comment by Parveshdhull** [comment]
maybe we should also improve [safe-merge](https://github.com/status-im/status-mobile/blob/9a60fc160070f76f9e02062344d07b616876a0ea/src/utils/re_frame.cljs#L42) function, so that it will convert `dispatch` into `dispatch-n`, instead of throwing error `merging-fx-with-common-keys`

**Comment by VolodLytvynenko** [comment]
Hi @Parveshdhull could you resolve the conflicts, please?

**Review by J-Son89 — APPROVED** [review]

File: src/status_im/events.cljs [hunk 1]
```diff
@@ -14,6 +14,7 @@
    status-im.chat.models.loading
    status-im.chat.models.transport
    [status-im.constants :as constants]
```
Comment (flexsurfer) [inline src/status_im/events.cljs]: oh thought we removed status-im.constants

**Comment by flexsurfer** [comment]
yeah this one is tricky one, just wondering what if won't reset root, in new code it should work, just use hot-reload trick instead

**Comment by Parveshdhull** [comment]
> yeah this one is tricky one, just wondering what if won't reset root, in new code it should work, just use hot-reload trick instead

we can try, but not sure if it will work: https://wix.github.io/react-native-navigation/docs/style-theme#changing-theme-dynamically

**Comment by flexsurfer** [comment]
> > yeah this one is tricky one, just wondering what if won't reset root, in new code it should work, just use hot-reload trick instead
> 
> we can try, but not sure if it will work: https://wix.github.io/react-native-navigation/docs/style-theme#changing-theme-dynamically

we don't use RNN themes, so it won't work, but hot reload might work, we just need to force re-render entire react tree

**Comment by Parveshdhull** [comment]
tried hot reload, its working (apart from status bar icon, bottom bar colors, (easily fixable)). This is neat idea :)

**Comment by Parveshdhull** [comment]
@VladimrLitvinenko, thank you for picking up PR. Please can you pause testing, I am thinking of adding few more changes to this PR. I will ping you once PR will be ready for testing.

**Comment by VolodLytvynenko** [comment]
> @VladimrLitvinenko, thank you for picking up PR. Please can you pause testing, I am thinking of adding few more changes to this PR. I will ping you once PR will be ready for testing.

Sure. Let me know when it is ready

**Comment by Parveshdhull** [comment]
Hi @VladimrLitvinenko, Thank you for your patience. PR is ready to be tested again.

Apart from #14780, please also test the behavior of status bar icon colors, and nav bar colors as per system theme changes and also app theme changes.
PR also remove the new-ui-toggle as it's not functional anymore, So please also test for regression for account creation and login.

### Known issue
These changes might create [this](https://github.com/status-im/status-mobile/issues/12921) distorted app state on old screens while changing theme. It's fine for now, we need to re-implement these screens without the RNN theme. But if distortion happening on new screens, please let me know.

**Review by flexsurfer — APPROVED** [review]
nice

**Review by jo-mut — APPROVED** [review]

**Comment by qoqobolo** [comment]
Hi @Parveshdhull , thanks for the fixes!
There is an issue with the status bar icon colors on both Android and iOS, take a look, please.

### ISSUE 1: Black font color of the status bar icons

**Case 1:**
1. Have your device in dark mode
2. Install Status 

In this case, the status bar icons have a black font with a black background from the beginning.

Screenshot 2023-01-23 at 12 35 38

**Case 2:**
1. Have your device in light mode
2. Install Status and create an account

You can see on the video, that in this case, the status bar looks okay during onboarding, but on the first `Jump to` screen icons get black with a black background.
The status bar gets also dark on other screens if you change the app theme to dark.

https://user-images.githubusercontent.com/67952253/214029504-23541d86-21a6-45d8-be90-9eb3cbc791fb.mp4


On 14 sec you can also see that the `Appearance` title stays light when changing theme, but looks like a part of this known issue with old screens:

> Known issue
These changes might create https://github.com/status-im/status-mobile/issues/12921 distorted app state on old screens while changing theme. It's fine for now, we need to re-implement these screens without the RNN theme. But if distortion happening on new screens, please let me know.

**Comment by Parveshdhull** [comment]
hi @qoqobolo, Thank you for testing the PR and finding this issue. Issue should be fixed now.

Also, yes Appearance color is expected as this still uses RNN theme.

**Comment by qoqobolo** [comment]
Thanks @Parveshdhull !

### ISSUE 2: The status bar icons get white on the white background (or vice versa) if switch the mode while a community channel is open

**Steps:**
1. Switch the theme to dark on your device
2. Install Status
3. Create a community
4. Open a channel
5. Switch the mode to light on your device while the channel is opened
**Actual result:** status bar gets white with white icons


https://user-images.githubusercontent.com/67952253/214089880-7558f654-8811-4010-ac29-cb0b73022429.mp4

---
Also, ISSUE 1 (Case 2) is still reproducible on iOS.

**Steps:**
1. Have your device in light mode
2. Install Status and go through the onboarding flow until the first `Jump to` screen
**Actual result:** fully black status bar


Screenshot 2023-01-23 at 17 15 24

**Comment by Parveshdhull** [comment]
Hi @qoqobolo,

**Issue 1:** changing the status bar color without the re-initializing of app, is currently not working in ios.
We can also reproduce this behavior by opening and closing the shell screen(press jump-to).
Status bar icon color will remain the same in the both home-stack and shell screen.
Somewhat related issue: https://github.com/wix/react-native-navigation/issues/5446

**Issue 2:** Currently we are not storing the ids of all opened screens, so on theme change we only change the status-bar icon color for the root screen. But once those screens closed and reopened the status bar color should be correct.

As Issue 1 looks a little not fixable(needs further research) and Issue 2 non blocker, I think we can skip them for now and log them separately. wdyt?

**Comment by Parveshdhull** [comment]
So, for ios, we are not able to change the status-bar icon color at runtime, but I made changes so that on login it should use the correct color.
**Before Login Screens:** Same as a system theme
**After Login Screens:** Same as app theme on most screens (except shell screen)

Please let me know if anything is not clear, It's a little confusing with so many edge cases. And thanks again for testing PR.

**Comment by qoqobolo** [comment]
> As Issue 1 looks a little not fixable(needs further research) and Issue 2 non blocker, I think we can skip them for now and log them separately. wdyt?

Sure, I'll log them in one new low-prio issue just to track them.
Thanks for the explanation! Overall, everything looks good to me.

But one more thing: e2e are failing because the `Enable Notifications` and `Welcome to Status` screens are back. Looks like they accidentally snuck into the code again :) Or it's on purpose? Can you clarify please?

https://user-images.githubusercontent.com/67952253/214276370-7dc9721f-052e-46ad-a156-5651f01896c7.mp4

**Comment by Parveshdhull** [comment]
> Looks like they accidentally snuck into the code again :) Or it's on purpose? Can you clarify please?

I also removed new-ui-toggle in PR and that accidentally exposed the else part(welcome screen).
Removed it, thank you for noticing.

**Comment by qoqobolo** [comment]
Thanks @Parveshdhull !
I don't see any other issues besides the ones mentioned.
PR can be merged.

**Pull request merged**